vermouth/processors/apply_rubber_band.py

```diff
@@ -39,29 +71,117 @@ def compute_force_constants(distance_matrix, lower_bound, upper_bound,
-    for (i, key_i),  (j, key_j) in itertools.combinations(enumerate(subgraph.nodes), 2):
-        shortest_path = len(nx.shortest_path(subgraph, key_i, key_j))
-        # The source and the target are counted in the shortest path
-        connectivity[i, j] = shortest_path <= separation + 2
-        connectivity[j, i] = connectivity[i, j]
+    for (i, key_i), (j, key_j) in itertools.combinations(enumerate(subgraph.nodes), 2):
+        try:
+            shortest_path = len(nx.shortest_path(subgraph, key_i, key_j))
+        except nx.NetworkXNoPath:
+            # There is no path between key_i and key_j so they are not
+            # connected; which is the default.
+            pass
+        else:
+            # The source and the target are counted in the shortest path
+            connectivity[i, j] = shortest_path <= separation + 2
+            connectivity[j, i] = connectivity[i, j]
     return connectivity
 
 
-
 def apply_rubber_band(molecule, selector,
                       lower_bound, upper_bound,
                       decay_factor, decay_power,
                       base_constant, minimum_force,
-                      bond_type):
+                      bond_type, res_min_dist=3):
+    r"""
+    Adds a rubber band elastic network to a molecule.
+
+    The eleastic network is applied as bounds between the atoms selected by the
+    function declared with the 'selector' argument. The equilibrium length for
+    the bonds is measured from the coordinates in the molecule, the force
+    constant is computed from the base force constant and an optional decay
+    function.
+
+    The decay function for the force constant is defined as:
+
+    .. math::
+
+        \exp^{-r(d - s)^p}
+
+    where :math:`r` is the decay rate given by the 'decay_factor' argument,
+    :math:`p` is the decay power given by 'decay_power', :math:`s` is a shift
+    given by 'lower_bound', and :math:`d` is the distance between the two atoms
+    in the molecule. If the rate or the power are set to 0, then the decay
+    function does not modify the force constant.
+
+    The 'selector' argument takes a callback that accepts a atom dictionary and
+    returns ``True`` if the atom match the conditions to be kept.
+
+    Parameters
+    ----------
+    molecule: Molecule
+        The molecule to which apply the elastic network. The molecule is
+        modified in-place.
+    selector: callback
+        Selection function.
+    lower_bound, upper_bound: float
+        The minimum and maximum length for a bond to be added, expressed in
+        nanometers.
+    decay_factor, decay_power: float
+        Parameters for the decay function.
+    base_constant: float
+        The base force constant for the bonds in kJ/mol/nm^2. If 'decay_factor'
+        or 'decay_power' is set to 0, then it will be the used force constant.
+    minimum_force: float
+        Minimum force constat in kJ/mol/nm^2 under which bonds are not kept.
```

Comment:
:math:? Or TeX formatting for the units? I'm pretty sure sphinx/rest can deal with it somehow.